wire-schema/src/test/java/com/squareup/wire/schema/PrunerTest.kt

```diff
@@ -1183,4 +1183,109 @@ class PrunerTest {
     assertThat(message.field("a")).isNotNull()
     assertThat(message.field("b")).isNull()
   }
+
+  @Test
+  fun excludingGoogleProtobufPrunesAllOptionsOnEnums() {
+    val schema = RepoBuilder()
+        .add("currency_code.proto", """
+             |package squareup;
+             |
+             |import "google/protobuf/descriptor.proto";
+             |
+             |message MessageOption {
+             |  optional string a = 1;
+             |  optional string b = 2;
+             |  optional string c = 3;
+             |}
+             |
+             |enum Style {
+             |  VERBOSE = 0;
+             |  POETIC = 1;
+             |  RUDE = 2;
+             |}
+             |
+             |extend google.protobuf.EnumValueOptions {
+             |  optional int32 max_length = 54000;
+             |  optional MessageOption message_option = 54001;
+             |  optional Style style = 54002;
+             |}
+             |
+             |enum Author {
+             |  ZEUS = 1 [(style) = POETIC, (max_length) = 12];
+             |  ARTEMIS = 2;
+             |  APOLLO = 3 [(style) = RUDE, (message_option) = {a: "some", c: "things"}];
+             |  POSEIDON = 4 [deprecated = true];
+             |}
+             """.trimMargin()
+        )
+        .schema()
+    val pruned = schema.prune(IdentifierSet.Builder()
+        .include("squareup.Author")
+        .exclude("google.protobuf.*")
+        .build())
+
+    assertThat(pruned.getType("squareup.Style")).isNull()
+    assertThat(pruned.getType("squareup.MessageOption")).isNull()
+    val authorType = pruned.getType("squareup.Author") as EnumType
+    assertThat(authorType.constant("ZEUS")!!.options.fields(null).size()).isZero()
```

Comment:
the null parameter here isn’t a beautiful API ... maybe provide a default value?